### PR TITLE
[QEMU] Fixing an unintentional variable shadowing

### PR DESCRIPTION
**Pull request by simar7**
This PR fixes the shadowing of the ```monitorPath``` variable
within the QEMU driver for Nomad. With this variable being shadowed
within the scope, all graceful shutdowns are effectively disabled
as ```monitorPath``` will not be set.

Also adds a helpful debug statement for the operator to verify.

Signed-off-by: Simarpreet Singh <simar@linux.com>

### CODE REVIEW

File: client/driver/qemu.go [hunk 1]
```diff
@@ -464,6 +464,7 @@ func (h *qemuHandle) Kill() error {
 	// If Nomad did not send a graceful shutdown signal, issue an interrupt to
 	// the qemu process as a last resort
 	if gracefulShutdownSent == false {
+		h.logger.Printf("[DEBUG] driver.qemu: graceful shutdown is not enabled, sending an interrupt signal to QEMU")
```
Comment (schmichael) [inline client/driver/qemu.go]: It's nice to include something to indicate what task log lines refer to. Perhaps something like:

> [DEBUG] driver.qemu: graceful shutdown disabled, sending interrupt signal to pid %d", h.userPid

**Comment by simar7** [comment]
@schmichael do we know if the tests are broken on master? The changes I've made don't seem to be related to the changes in the PR.

**Comment by dadgar** [comment]
@simar7 Can we improve the test to check that the vm shutdown post signaling? The test should fail before this change and pass after.

**Comment by cheeseprocedure** [comment]
Chiming in, as I investigated validation of VM shutdown as part of [PR# 3411](https://github.com/hashicorp/nomad/pull/3411): the minimal Linux image in [linux-0.2.img](https://github.com/hashicorp/nomad/tree/master/client/driver/test-resources/qemu) doesn't appear to respond to an ACPI shutdown signal passed via the QEMU monitor, which makes a full test more difficult. (This can surely be fixed, but I haven't dug any deeper.)

**Comment by schmichael** [comment]
@cheeseprocedure @simar7 I tried adding an Alpine Linux image in #3819 to test graceful shutdowns and have had no luck. Not sure what I'm doing wrong.

(I pushed it to my own fork to keep the 40MB image out of the main Nomad repo if this effort doesn't work out.)

**Comment by cheeseprocedure** [comment]
@schmichael it looks like ``TestQemuDriver_GracefulShutdown`` does pass if you add a significant ``time.Sleep`` after confirming the presence of the monitor socket. I'm guessing this is because the guest has to reach a state where it's ready to actually receive the ACPI-poweroff signal.

I'm not aware of a sensible way to verify the state of the guest in a black-box fashion before triggering a shutdown. Perhaps we could call ``sendQemuShutdown`` on a ticker until the test times out or the process exits?

**Comment by schmichael** [comment]
@cheeseprocedure Thanks for the tip! The Alpine image has sshd installed, so I just poll until sshd starts and then know it's safe to issue the shutdown command. Pushed updates to #3819.

Looks like everything is working!

**Pull request merged**